File: service-common/src/main/java/packages/TopPackage.java
package packages;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import ru.cristalix.core.network.CorePackage;
import tops.PlayerTopEntry;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@RequiredArgsConstructor
public class TopPackage extends CorePackage {

    // request
    private final TopType topType;
    private final int limit;

    // response
    private List<PlayerTopEntry<Object>> entries;

    public enum TopType {

        WINS,
        GAMES
        ;

    }

}

File: common/src/main/kotlin/me/reidj/common/package/TopPackage.kt
package me.reidj.common.`package`

import me.reidj.common.top.PlayerTopEntry


data class TopPackage(val type: TopType, val limit: Int): BridgePackage() {
    lateinit var entries: List<PlayerTopEntry<Any>>

    enum class TopType {
        WINS,
        ;
    }
}
